### PR TITLE
[feature](parquet-reader) add rle bool and delta decoder to read AWS Glue

### PR DESCRIPTION
**Pull request by wsjz**
# Proposed changes

Support delta encoding and rle(bool) to read Glue data
add delta bit pack decoder,
add delta length byte array decoder,
add delta byte array decoder.
add rle bool decoder.

We find some data type is read with delta encoding on AWS Glue, so it should be supported.
The definition of delta encoding can refer to the [delta encoding in parquet](https://github.com/apache/parquet-format/blob/master/Encodings.md#delta-encoding-delta_binary_packed--5).

## Problem summary

Describe your changes.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/vec/exec/format/parquet/delta_bit_pack_decoder.h [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+template <typename T>
+class DeltaBitPackDecoder final : public Decoder {
+public:
+    using UT = std::make_unsigned_t<T>;
+
+    DeltaBitPackDecoder() = default;
+    ~DeltaBitPackDecoder() override = default;
+    Status decode_values(MutableColumnPtr& doris_column, DataTypePtr& data_type,
+                         ColumnSelectVector& select_vector) override {
+        size_t num_values = select_vector.num_values();
```
Comment (AshinGau) [inline be/src/vec/exec/format/parquet/delta_bit_pack_decoder.h:42]: How about null values.

**Comment by wsjz** [comment]
run build all

**Comment by morningman** [comment]
run buildall

File: be/src/vec/exec/format/parquet/fix_length_plain_decoder.cpp [hunk 2]
```diff
@@ -111,7 +111,8 @@ Status FixLengthPlainDecoder::decode_values(MutableColumnPtr& doris_column, Data
     case TypeIndex::String:
         [[fallthrough]];
     case TypeIndex::FixedString:
-        if (_physical_type == tparquet::Type::FIXED_LEN_BYTE_ARRAY) {
+        if (_physical_type == tparquet::Type::FIXED_LEN_BYTE_ARRAY ||
+            _physical_type == tparquet::Type::BYTE_ARRAY) {
```
[the remark this answers is not in the document]
Comment (wsjz) [reply be/src/vec/exec/format/parquet/fix_length_plain_decoder.cpp:113]: Move to case String

File: be/src/vec/exec/format/parquet/delta_bit_pack_decoder.h [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+        int num_valid_values;
+        RETURN_IF_ERROR(_get_internal(_values.data(), num_values - null_count, &num_valid_values));
+        DCHECK_EQ(num_values - null_count, num_valid_values);
+
+        _data = _values.data();
+        _type_length = sizeof(_field_schema->parquet_schema.type_length);
+        _data->size = _values.size() * _type_length;
+        init_values_converter();
+
+        return _fix_plain_decoder->decode_values(doris_column, data_type, select_vector);
```
[the remark this answers is not in the document]
Comment (wsjz) [reply be/src/vec/exec/format/parquet/delta_bit_pack_decoder.h:220]: But we still need to decode numeric and decode date in ByteArrayPlainDecoder, it now unsupported.

File: be/src/vec/exec/format/parquet/delta_bit_pack_decoder.cpp [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+    max_values = std::min(max_values, _num_valid_values);
+    if (max_values == 0) {
+        *out_num_values = max_values;
+        return Status::OK();
+    }
+
+    int suffix_read;
+    RETURN_IF_ERROR(_suffix_decoder.decode(buffer, max_values, &suffix_read));
+    if (PREDICT_FALSE(suffix_read != max_values)) {
+        return Status::EndOfFile("Read " + std::to_string(suffix_read) + ", expecting " +
```
Comment (AshinGau) [inline be/src/vec/exec/format/parquet/delta_bit_pack_decoder.cpp:222]: please remove all `EndOfFile`, and use `Error("{}, {}, {}", str1, str2, str3)` style.

**Comment by wsjz** [comment]
wait for regression case

**Comment by wsjz** [comment]
run buildall

**Comment by wsjz** [comment]
run buildall

**Comment by wsjz** [comment]
run p0

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**